packages/google_maps_flutter/google_maps_flutter_android/android/src/main/java/io/flutter/plugins/googlemaps/cozy/CozyMarkerBuilder.java

```diff
@@ -98,17 +104,38 @@ private Bitmap getMarkerBitmapFromElements(CozyMarkerElements cozyElements){
         canvas.drawPath(bubblePath, fillPaint);
         canvas.drawPath(bubblePath, strokePaint);
 
-       
-        // draws the text
-        for (CozyMarkerElement label : labels){
-            String text = (String) label.data;
+        // draws the bubbles apart from the marker one
+        if (bubbles.length > 1) {
```

Comment:
I think that instead of using a list of bubbles, which can be really confusing (list doesn't provide a good contract as we can see here checking the length), we could add a new object named like `counterBubble`. Although they both describe similar geometric shapes the main goal of `CozyMarkerElements` contract was to provide more an easy way to understand which elements compose the marker than to describe a list of geometrical entities.

I think the responsibility to decide what to do to those geometrical entities (if they are bubble or not) must be left to the `cozyMarkerBuilder` itself.

Similar thing with label: my goal was to have just the single label of the marker on the `labels` object. The only reason for that to be a list is because of implementation details of animation (that must have "two" labels fading-in and fading-out simutaneously). For the counter label, I would suggest to use the `data` of the bubble, or create another `CozyMarkerElement`, because the indexes could really mix-up on the animations and introduce unintended bugs.

Reply:
But I liked the idea to generalize the geometric entity and the cozyMarkerBuilder just render it without duplicating code, maybe we could have a type inside the `CozyMarkerElement` or create subclasses to differ between bubble and label, and the labels of counter and bubble are instances of those types